### PR TITLE
Add test for ResolveVersion

### PR DESCRIPTION
**Pull request by TheoBrigitte**
Towards: https://github.com/giantswarm/giantswarm/issues/6844

This PR implements test for the `ResolveVersion` method which resolve a git reference and find the project version for it.

`gitrepo` uses itself for test purpose.

There are 3 version to test:
- untagged version: `v0.0.0-2e7604b8b3806b20ff305eb4e1a852c784ba34ca`
- tagged version: `test`
- above tagged version: `test-1313262ada5f23ebf83ee3d2b6af4b56f16dc0c1`

And 3 reference to test:
- commit sha
- branch
- tag

Note: last test case is to be set.

### CODE REVIEW

File: pkg/gitrepo/repo_test.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+		},
+		{
+			// tag reference
+			Ref:             "test-tag",
+			ExpectedVersion: "test-tag",
+		},
+	}
+
+	c := Config{
+		AuthBasicToken: os.Getenv("GITHUB_BOT_TOKEN"),
```
Comment (kopiczko) [inline pkg/gitrepo/repo_test.go]:
```suggestion
```

File: pkg/gitrepo/repo_test.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+	"context"
+	"os"
+	"testing"
+)
+
+// TestResolveVersion tests ResolveVersion method which resolve a git reference and find the project version for it.
+func TestResolveVersion(t *testing.T) {
+	testCases := []struct {
+		Ref             string
+		ExpectedVersion string
```
Comment (kopiczko) [inline pkg/gitrepo/repo_test.go]: https://github.com/giantswarm/fmt/blob/master/go/unit_tests.md

```
		name            string
		inputRef        string
		expectedVersion string
```

Comment (TheoBrigitte) [reply pkg/gitrepo/repo_test.go]: solved

File: pkg/gitrepo/repo_test.go (hunk 1 above)
Comment (TheoBrigitte) [reply pkg/gitrepo/repo_test.go]: removed

**Comment by TheoBrigitte** [comment]
Now using `v0.1.0` tag for testing as semver tags are required.

File: pkg/gitrepo/repo.go [hunk 3]
```diff
@@ -16,6 +18,8 @@ import (
 	"gopkg.in/src-d/go-git.v4/storage/filesystem"
 )
 
+var tagRegex = regexp.MustCompile(`^v?[0-9]+\.[0-9]+\.[0-9]+`)
```
Comment (kopiczko) [inline pkg/gitrepo/repo.go]:
```suggestion
var tagRegex = regexp.MustCompile(`^v[0-9]+\.[0-9]+\.[0-9]+`)
```

File: pkg/gitrepo/repo_test.go [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+		{
+			name:            "case 3: branch reference",
+			inputRef:        "test-branch",
+			expectedVersion: "0.1.0",
+		},
+		{
+			name:            "case 4: tag reference",
+			inputRef:        "v0.1.0",
+			expectedVersion: "0.1.0",
+		},
```
Comment (kopiczko) [inline pkg/gitrepo/repo_test.go]: I think this case should go on top :)

Comment (TheoBrigitte) [reply pkg/gitrepo/repo_test.go]: done :)

**Pull request merged**